smarts/core/scenario.py

```diff
@@ -77,13 +75,13 @@ class Scenario:
     def __init__(
         self,
         scenario_root: str,
-        route: str = None,
-        missions: Dict[str, Mission] = None,
-        social_agents: Dict[str, SocialAgent] = None,
-        log_dir: str = None,
+        route: Optional[str] = None,
+        missions: Optional[Dict[str, Mission]] = None,
+        social_agents: Optional[Dict[str, SocialAgent]] = None,
+        log_dir: Optional[str] = None,
         surface_patches: list = None,
```

Comment:
Why not `Optional` for `surface_patches` too?  (Maybe it could even be `Optional[Sequence[Dict[str, Any]]]`.)

Reply:
The pytype throws hundreds of type errors. Not all type errors have been fixed yet. This pull request is an incremental fix.

I have included your suggested type for `surface_patches` now.